scripts/jenkins/test.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -e
+export GOPATH=$WORKSPACE
+eval "$(gvm $GO_VERSION)"
+go get -v -u github.com/jstemmer/go-junit-report
+go get -v -u github.com/axw/gocov/gocov
+go get -v -u github.com/matm/gocov-html
+go get -v -u github.com/axw/gocov/...
+go get -v -u github.com/AlekSi/gocov-xml
+
+go get -v -t ./...
+
+export COV_FILE="build/coverage.cov"
```

Comment:
I don't think we need coverage for both this and the docker tests. Having it for the docker tests should be enough, since those tests are a superset.

Reply:
this one thing I have to ask you, there are differences between two coverage reports, the docker one has more files, these are they:

```
go.elastic.co/apm/apmtest/httpsuite.go
go.elastic.co/apm/apmtest/testlogger.go
go.elastic.co/apm/apmtest/withtransaction.go
go.elastic.co/apm/internal/apmcontext/context.go
go.elastic.co/apm/internal/apmdebug/debug.go
go.elastic.co/apm/internal/apmschema/schema.go
go.elastic.co/apm/internal/radix/radix.go
go.elastic.co/apm/transport/transporttest/err.go
go.elastic.co/apm/transport/transporttest/recorder.go
```

What is it the good coverage test?
I did not check the unit test executed but probably are different too.

Reply:
It's because of the "-coverpkg" flag.

`go test -cover ./...` will run the tests for each package in subdirectories of `./`, but for each package it only includes coverage information of its own package. i.e. test coverage for `go.elastic.co/apm/module/apmhttp` only contains coverage information for `go.elastic.co/apm/module/apmhttp`.

In the docker case, it runs "make coverage", which runs scripts/test_coverage.sh. In that script you can see we add a flag, `-coverpkg=go.elastic.co/apm/...`. This instructs the go command to include coverage for all packages under `go.elastic.co/apm/...`

That script is slightly more complex than needed here, since it also works with older versions of Go. You could just run `go test ./... -v -coverprofile="${COV_FILE}" -coverpkg=go.elastic.co/apm/...` to get the same effect, for recent versions of Go.

Reply:
so if we add `-coverpkg=go.elastic.co/apm/...` to the test.sh (test stage) both will be the same, ok, in that case the docker test can be removed.

Reply:
> so if we add -coverpkg=go.elastic.co/apm/... to the test.sh (test stage) both will be the same, ok, in that case the docker test can be removed.

They might cover the same files, but not the same lines. The docker tests run a super-set of tests: integration tests that only run when certain environment variables are set (e.g. PGHOST). The docker one should be kept, and this one removed.

Reply:
gotcha, it is the other way around